kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -503,6 +533,18 @@ namespace Kratos
         }
     }
 
+    template<std::size_t TDim>
+    void CalculateDiscontinuousDistanceToSkinProcess<TDim>::ReplaceZeroDistances(
+        Vector& rElementalDistances)
+    {
+        const double multiplicator = (mOptions.Is(CalculateDiscontinuousDistanceToSkinProcessFlags::USE_NEGATIVE_EPSILON_FOR_ZERO_VALUES))  ? -1e3 : 1e3;
+        for (auto& r_distance : rElementalDistances) {
+            if (std::abs(r_distance) < std::numeric_limits<double>::epsilon()) {
+                r_distance = multiplicator*std::numeric_limits<double>::epsilon();
+            }
```

Comment:
> 1. Distance values of 0 are NOT valid from the distance algorithms calculation perspective. The treatment of these should be done inside this process. This can be achieved by adding an signed epsilon to the zero values.

Here 0 values are replaced by an epsilon that is `1e3*std::numeric_limits<double>::epsilon()`

Should this factor by user defined or is it fine leaving it hard-coded?

Note that the 'TO_SPLIT' elements are checked using `std::numeric_limits<double>::epsilon()`. This ensures that an element with all distance values positive except a 0.0 intersected point, will not count as intersected. If all distance values are negative except a 0.0 intersection, the element will be counted as intersected. 

The sign of the epsilon can be controlled by the flag: `USE_NEGATIVE_EPSILON_FOR_ZERO_VALUES`, changing the above behaviour.